### PR TITLE
Fxa fixes

### PR DESCRIPTION
**Pull request by jrconlin**

### CODE REVIEW

File: autoendpoint/src/middleware/sentry.rs [hunk 1]
```diff
@@ -35,6 +35,11 @@ pub fn sentry_middleware(
                 if let Some(api_err) = error.as_error::<ApiError>() {
                     // if it's not reportable, and we have access to the metrics, record it as a metric.
                     if !api_err.kind.is_sentry_event() {
+                        // The error (e.g. VapidErrorKind::InvalidKey(String)) might be too cardinal,
+                        // but we may need that information to debug a production issue. We can
+                        // add an info here, temporarily turn on info level debugging on a given server,
+                        // capture it, and then turn it off before we run out of money.
```
Comment (pjenvey) [inline autoendpoint/src/middleware/sentry.rs:41]: We really need to just add extras support, logged: https://mozilla-hub.atlassian.net/browse/CONSVC-1886

File: autoendpoint/src/error.rs [hunk 2]
```diff
@@ -58,6 +58,9 @@ pub enum ApiErrorKind {
     #[error(transparent)]
     PayloadError(actix_web::Error),
 
+    #[error("Payload Error: {0}")]
+    PayloadErrorS(String),
```
Comment (pjenvey) [inline autoendpoint/src/error.rs:62]: Do we need this? I don't see it being used

File: autoendpoint/src/middleware/sentry.rs (hunk 1 above)
Comment (jrconlin) [reply autoendpoint/src/middleware/sentry.rs:41]: Ah, good catch! I missed that.

File: autoendpoint/src/error.rs (hunk 2 above)
Comment (jrconlin) [reply autoendpoint/src/error.rs]: Ah, good catch! I missed that.

File: autoendpoint/src/extractors/subscription.rs [hunk 3]
```diff
@@ -292,7 +306,8 @@ mod tests {
             base64::STANDARD,
         )
         .unwrap();
-        let public_key = "BM3bVjW_wuZC54alIbqjTbaBNtthriVtdZlchOyOSdbVYeYQu2i5inJdft7jUWIAy4O9xHBbY196Gf-1odb8hds".to_owned();
+        // Specify a potentially invalid padding.
```
Comment (pjenvey) [inline autoendpoint/src/extractors/subscription.rs:309]: Might be worth it to make this a separate test case

**Review by pjenvey — APPROVED** [review]
Looks good but maybe the padding case could be a separate test case

**Pull request merged**